jax/_src/scipy/sparse/linalg.py

```diff
@@ -483,13 +484,13 @@ def body_fun(value):
 
   initialization = (x0, 0, unit_residual, residual_norm)
   x_final, k, _, err = lax.while_loop(cond_fun, body_fun, initialization)
-  _ = k # Until we can pass this out
+  _ = k  # Until we can pass this out
   _ = err
   return x_final  # , info
 
 
-def _gmres(A, b, x0=None, *, tol=1e-5, atol=0.0, restart=20, maxiter=None,
-           M=None, qr_mode=False):
+def gmres(A, b, x0=None, *, tol=1e-5, atol=0.0, restart=20, maxiter=None,
+          M=None, solve_method='incremental'):
```

Comment:
Given the benchmark results you just shared, would it maybe be better to use the "direct" method as a default? It seems like the direct method is close enough on CPU while significantly faster on GPU. Granted those were very synthetic benchmarks but I think most use cases won't be all that different. Also, I suspect that the users most likely to care are also the ones most likely to override the default behavior.